<html xmlns:concordion="http://www.concordion.org/2007/concordion">
<head>
    <link href="../../../../concordion.css" rel="stylesheet" type="text/css" />
    <title>"Continue after exceptions" option</title>
</head>
<body>

    <h1>"Continue after exceptions" option</h1>

    <p>
        By default, exceptions are reported but Concordion continues
        processing the rest of the test, so it can show all the problems
        not just the first one. Often you'll have a page of green with
        one red item. If it stopped processing on the red item, you 
        wouldn't get the feedback that everything else passed.
    </p>

    <p> However, <a href="FailFast.html">fixtures can be annotated with <code>@FailFast</code></a> to stop further
        processing if an exception is encountered.
    </p>
    

    <div class="example">

        <h3>Example</h3>

        <p>When the fixture is not annotated with <code>@FailFast</code> and processes the following:</p>

<pre class="html" concordion:set="#fragment">
&lt;p concordion:execute="myExceptionThrower()"&gt;
This method throws an exception.
&lt;/p&gt;

&lt;p concordion:execute="myMethod()"&gt;
This method executes quietly.
&lt;/p&gt;
</pre>

        <p>
            It <span concordion:assertEquals="myMethodWasCalledProcessing(#fragment) ? 'calls' : 'does not call'">calls</span>
            <code>myMethod()</code> in the Java fixture code because the exception thrown by the
            first method call does not stop further processing of the test.
        </p>

    </div>


</body>
</html>